cmd/balance-agent/main.go

```diff
@@ -57,16 +58,41 @@ var (
 func getData(ctx *cli.Context) (*entities.Data, error) {
 
 	resp := &entities.Data{}
+	resp.Endpoint = ctx.String("rpcserver")
+	resp.ApiType = nil
 
-	if ctx.Bool("userest") {
-		v := int(api.LND_REST)
-		resp.ApiType = &v
-	} else {
-		v := int(api.LND_GRPC)
+	// Assume CLN first
+	if !ctx.Bool("ignorecln") {
```

Comment:
this is getting quite complex

Reply:
Refactored to be a bit more readable now